lib/vdc/BOVCollection.cpp

```diff
@@ -0,0 +1,368 @@
+            SetErrMsg(message.c_str());
+            return -1;    // Bad token
+        }
+
+        if (lineStream.bad()) {
+            value.clear();
+            std::cout << "FAIL v " << token << std::endl;
+            std::string message = "Invalid value for " + token + " in BOV header file.";
+            SetErrMsg(message.c_str());
+            return -1;    // Bad token
+        }
+
+        if (verbose) {
+            std::cout << std::setw(20) << token << " ";
+            for (int i = 0; i < value.size(); i++) std::cout << value[i] << " ";
+            std::cout << std::endl;
+        }
+        return 1;    // we now have a token
+    }
+    return 0;    // we didn't find the token
+}
+
+void BOVCollection::_findTokenValue(std::string &line) const
+{
+    std::string delimiter = ": ";
+
+    size_t      pos = 0;
+    std::string token;
+    while ((pos = line.find(delimiter)) != std::string::npos) {
+        token = line.substr(0, pos);
+        line.erase(0, pos + delimiter.length());
+    }
+}
+
+size_t BOVCollection::_sizeOfFormat(DC::XType type) const
+{
+    switch (type) {
+    case DC::XType::INT32: return 4;
+    case DC::XType::FLOAT: return 4;
+    case DC::XType::DOUBLE: return 8;
+    default: return -1;
+    }
+}
+
+void BOVCollection::_swapBytes(void *vptr, size_t size, size_t n) const
+{
+    unsigned char *ucptr = (unsigned char *)vptr;
+    unsigned char  uc;
+    size_t         i, j;
+
+    for (j = 0; j < n; j++) {
+        for (i = 0; i < size / 2; i++) {
+            uc = ucptr[i];
+            ucptr[i] = ucptr[size - i - 1];
+            ucptr[size - i - 1] = uc;
+        }
+        ucptr += size;
+    }
+}
+
+template<class T> int BOVCollection::ReadRegion(const std::vector<size_t> &min, const std::vector<size_t> &max, T region)
+{
+    FILE *fp = fopen(_dataFile.c_str(), "rb");
+    if (!fp) {
+        SetErrMsg("Invalid file: %d", fp);
+        return -1;
+    }
+
+    size_t formatSize = _sizeOfFormat(_dataFormat);
+    if (formatSize < 0) {
+        SetErrMsg("Unspecified data format");
+        return (-1);
+    }
+
+    if (_dataSize.size() != 3) {
+        SetErrMsg("Invalid grid size (must be 3D)");
+        return (-1);
+    }
+    size_t numValues = _dataSize[0] * _dataSize[1] * _dataSize[2];
+
+    int  n = 1;
+    bool systemLittleEndian = *(char *)&n == 1 ? true : false;
+    bool dataLittleEndian = _dataEndian == "LITTLE" ? true : false;
+    bool needSwap = systemLittleEndian != dataLittleEndian ? true : false;
+
+    // Read a "pencil" of data along the X axis, one row at a time
+    size_t count = max[0] - min[0] + 1;
+    for (int k = min[2]; k <= max[2]; k++) {
+        int zOffset = _dataSize[0] * _dataSize[1] * k;
+        for (int j = min[1]; j <= max[1]; j++) {
+            int xOffset = min[0];
+            int yOffset = _dataSize[0] * j;
+            int offset = formatSize * (xOffset + yOffset + zOffset);
+
+            unsigned char readBuffer[count * formatSize];
```

Comment:
Please use a heap allocation instead of a stack allocation here.

Reply:
Fixed.

Reply:
Did you push new changes? Seems stack allocation still.

Reply:
I think we don't see it here because our comments are attached to the old file.  The permalink below shows the current version of BOVCollection.cpp where a WASP::SmartBuf is being used.

https://github.com/NCAR/VAPOR/blob/c65794fb6d0c2159f111b2cb99544b8a19efa670/lib/vdc/BOVCollection.cpp#L331

Reply:
@sgpearse the SmartBuf pre-dates C++11 and should probably be deprecated. I typically just use a pre-allocated STL vector and access it via the 'data' member. E.g.

```
std::vector<float> buf (buf_size);
float *fptr = buf.data()
```

Reply:
No problem.  Fixed.